src/Azure.Functions.Cli/Actions/KubernetesActions/KubernetesDeployAction.cs

```diff
@@ -58,8 +62,11 @@ public override ICommandLineParserResult ParseArgs(string[] args)
             SetFlag<int>("cooldown-period", "The cooldown period for the deployment before scaling back to 0 after all triggers are no longer active. Default: 300 (seconds)", p => CooldownPeriod = p);
             SetFlag<int>("min-replicas", "Minimum replica count", m => MinReplicaCount = m);
             SetFlag<int>("max-replicas", "Maximum replica count to scale to by HPA", m => MaxReplicaCount = m);
-            SetFlag<string>("secret-name", "The name of a kubernetes secret collection to use in the deployment instead of generating one based on local.settings.json", sn => SecretsCollectionName = sn);
-            SetFlag<string>("config-map-name", "The name of a config map to use in the deployment", cm => ConfigMapName = cm);
+            SetFlag<string>("funcappkeys-secret-name", "The name of a kubernetes secret collection to use for the function app keys (host keys, function keys etc.)", ksn => FunAppKeysSecretsCollectionName = ksn);
+            SetFlag<string>("funcappkeys-config-map-name", "The name of a kubernetes config map to use for the function app keys (host keys, function keys etc.)", kcm => FunAppKeysConfigMapName = kcm);
+            SetFlag<bool>("mount-funcappkeys-as-containervolume", "The flag indicating to mount the func app keys as container volume", kmv => MountFuncAppKeysAsContainerVolume = kmv);
+            SetFlag<string>("appsettings-secret-name", "The name of a existing func app settings kubernetes secret collection to use in the deployment instead of generating one based on local.settings.json", sn => AppSettingsSecretsCollectionName = sn);
+            SetFlag<string>("appsettings-config-map-name", "The name of an existing config map with func app settings to use in the deployment", cm => AppSettingsConfigMapName = cm);
```

Comment:
I'd rather not rename existing flags if possible. I think we can get rid of `funcappkeys-config-map-name`, and rename `funcappkeys-secret-name` to `keys-secret-name`, then leave `secret-name` and `config-map-name` as is